gold/gold_matrix_plot_D.m

```diff
@@ -1,15 +1,22 @@
-%% Generate gold data for C++ testing of matrix_plot_D component
-data_dir = strcat("generated-inputs", "/", "matrix_plot_D", "/");
+gen_realspace_transform( continuity1, 'continuity_1', 'lev', 8, 'deg', 7);
```

Comment:
I'm unfamiliar with this `gen_realspace_transform` function? Where is it? Also, I'm not sure why this PR needs to modify any other gold data generation routine other than the new `gold_wavelet_to_realspace`?

Reply:
Gen realspace transform is the c++ equivalent of matrix plot D. This function is called from wavelet to realspace and is tested separately.

I actually pulled these changes into my pr, but you can merge them and I'll rebase if you'd prefer to keep them separate.

Reply:
@bmcdanie which PR are you referring to? Are you suggesting these changes are already merged in?

Reply:
yes - I believe they are. this one had been sitting a while, so I updated the file to produce the gold values we test on and added it to a previous MR.

Reply:
This PR needed to modify 3 gold data generation routines I believe - 
1. gold_wavelet_to_realspace
2. gold_matrix_plot_D
3. make_all_gold

The changes in the remaining 6 files consist of added whitespace for readability, removal of trailing hidden characters at the ends of lines, and and autogenerated file in the two_scale directory